source/parser.hera

```diff
@@ -982,10 +981,10 @@ AfterReturnShorthand
 # The parameters in a function definition
 Parameters
   NonEmptyParameters
-  "" ->
+  TypeParameters?:tp Loc:p ->
```

Comment:
This changes the current input: `<div>=> something`
which currently compiles to `<div>{() => something}</div>` but now gets treated as a parameterized arrow function

But perhaps that JSX is not something that should have been supported in the first place, because it shadows a TypeScript arrow function. In fact, `<For>(item) => foo` already wasn't supported as JSX; the function had to start on the next line.

For symmetry, though, I think this JSX should probably be supported, and currently isn't with this PR:
```coffee
<div>
  => something
```
Perhaps another `__` that should be a `_?`?

Also, `<T>=> x` and `<T>(x) => x` currently output arrow functions starting with `<T>` but that's not valid TSX. The output should start with `<T,>`.  I thought we fixed this, but I can't find any code that does it...  We ought to do it for both `TypeParameters` and `TypeArguments`, when there's one parameter that's just a single word.